surefire-api/src/main/java/org/apache/maven/surefire/api/testset/TestListResolver.java

```diff
@@ -514,4 +519,120 @@ private static boolean haveMethodPatterns( Set<ResolvedTest> patterns )
         }
         return false;
     }
+
+    private boolean methodMatchCheck( ResolvedTest includedPattern, String methodName )
+    {
+        MethodMatcher methodMatcher = new MethodMatcher();
```

Comment:
You don't need this helper method. Just create methodMatcher before calling this method and call the methodMatcher with the includedPattern and methodName